lib/ansible/modules/messaging/rabbitmq_binding.py

```diff
@@ -116,119 +116,113 @@
 
-    if r.status_code==200:
+    if r.status_code == 200:
         binding_exists = True
         response = r.json()
-    elif r.status_code==404:
+    elif r.status_code == 404:
         binding_exists = False
         response = r.text
     else:
         module.fail_json(
-            msg = "Invalid response from RESTAPI when trying to check if exchange exists",
-            details = r.text
+            msg="Invalid response from RESTAPI when trying to check if exchange exists",
+            details=r.text
         )
 
-    if module.params['state']=='present':
+    if module.params['state'] == 'present':
         change_required = not binding_exists
     else:
         change_required = binding_exists
 
     # Exit if check_mode
     if module.check_mode:
-        module.exit_json(
-            changed= change_required,
-            name = module.params['name'],
-            details = response,
-            arguments = module.params['arguments']
-        )
+        result['changed'] = change_required
+        result['details'] = response
+        result['arguments'] = module.params['arguments']
+        module.exit_json(**result)
 
     # Do changes
     if change_required:
         if module.params['state'] == 'present':
-            url = "http://%s:%s/api/bindings/%s/e/%s/%s/%s" % (
-                module.params['login_host'],
-                module.params['login_port'],
-                urllib.quote(module.params['vhost'],''),
-                urllib.quote(module.params['name'],''),
+            url = "%s/%s/e/%s/%s/%s" % (
+                base_url,
+                urllib.quote(module.params['vhost'], ''),
+                urllib.quote(module.params['name'], ''),
                 dest_type,
-                urllib.quote(module.params['destination'],'')
+                urllib.quote(module.params['destination'], '')
             )
 
             r = requests.post(
                 url,
-                auth = (module.params['login_user'],module.params['login_password']),
-                headers = { "content-type": "application/json"},
-                data = json.dumps({
+                auth=(module.params['login_user'], module.params['login_password']),
+                headers={"content-type": "application/json"},
+                data=json.dumps({
                     "routing_key": module.params['routing_key'],
                     "arguments": module.params['arguments']
-                    })
+                })
             )
         elif module.params['state'] == 'absent':
-            r = requests.delete( url, auth = (module.params['login_user'],module.params['login_password']))
+            r = requests.delete(url, auth=(module.params['login_user'], module.params['login_password']))
 
         if r.status_code == 204 or r.status_code == 201:
-            module.exit_json(
-                changed = True,
-                name = module.params['name'],
```

Comment:
Are you referring [this](https://github.com/ansible/ansible/pull/24590/files/aee05ea2ae7c5e7cdd84e9037c080b39f0f8e66e#diff-d33fa8ead78fe967669bddbe3e490f6fR140)